CMakeLists.txt

```diff
@@ -213,13 +213,15 @@ if(BUILD_STATIC)
 endif()
 
 # Check dependencies
-if(NOT TARGET_GLES OR TARGET_DESKTOP_GLES)
-    set(OpenGL_GL_PREFERENCE GLVND) # since CMake 3.11
-    find_package(OpenGL REQUIRED)
-elseif(TARGET_GLES2)
-    find_package(OpenGLES2 REQUIRED)
-else()
-    find_package(OpenGLES3 REQUIRED)
+if(WITH_GL OR TARGET_GL)
```

Comment:
After thinking about this a bit more, this should be just `if(WITH_GL)`. `TARGET_GL` can be only enabled if `WITH_GL` is enabled.